### PR TITLE
ATB-1265 - AIS: Added feature tests for User Account in Suspended and Blocked state

### PR DESCRIPTION
**Pull request by ChrisBates1**
### What changed

Feature tests added for a Account with Intervention state (suspended and blocked)

### Issue tracking

- [ATB-1265](https://govukverify.atlassian.net/browse/ATB-1265)

## Testing

All tests pass in Dev locally:
image

## Checklists
- [x] Did not commit any not-required changes to the `src/infra/**/samconfig.toml`
- [x] Tested changes and included test evidence in the PR, if appropriate
- [x] Included all required tags and other properties for any new resources in the SAM template
- [x] Ensured that any new resources in the SAM Template follow appropriate naming conventions
- [x] Ensured that naming of new resources is compatible with deploying multiple stacks with custom stack names during development
- [x] Ensured that no log lines include PII or other sensitive data
- [x] Implemented unit testing for any new logic implemented, if appropriate
- [x] Ensured that all commits in this PR are signed
- [x] Ensured appropriate code coverage is maintained by unit tests

[ATB-1265]: https://govukverify.atlassian.net/browse/ATB-1265?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by liamquayle — APPROVED** [review]

File: feature-tests/tests/resources/step-definitions/invoke-api-gateway-happy-path.step.ts [hunk 1]
```diff
@@ -52,6 +52,86 @@ defineFeature(feature, (test) => {
...
+  test('Happy Path - Suspended Account - Get Request to /ais/userId - Returns Expected Data for <aisEventType>', ({
+    given,
+    when,
+    then,
+  }) => {
+    given(
+      /^I send an (.*) intervention message to the TxMA ingress SQS queue for a Account in (.*) state$/,
+      async (originalAaisEventType, aisEventType) => {
+        await sendSQSEvent(testUserId, originalAaisEventType);
+        await sendSQSEvent(testUserId, aisEventType);
```
Comment (chiarapascucci-gds) [inline feature-tests/tests/resources/step-definitions/invoke-api-gateway-happy-path.step.ts:64]: Could it be maybe prudent to put a wait between these two steps?

File: feature-tests/tests/resources/step-definitions/invoke-api-gateway-happy-path.step.ts [hunk 2]
```diff
@@ -52,6 +52,86 @@ defineFeature(feature, (test) => {
...
+  test('Happy Path - Blocked Account - Get Request to /ais/userId - Returns Expected Data for <aisEventType>', ({
+    given,
+    when,
+    then,
+  }) => {
+    given(
+      /^I send an (.*) intervention message to the TxMA ingress SQS queue for a Account in (.*) state$/,
+      async (originalAaisEventType, aisEventType) => {
+        await sendSQSEvent(testUserId, originalAaisEventType);
+        await sendSQSEvent(testUserId, aisEventType);
```
Comment (chiarapascucci-gds) [inline feature-tests/tests/resources/step-definitions/invoke-api-gateway-happy-path.step.ts:104]: as above, may be good to wait between sending the messages here.. just so we can be more confident they are processed in the order we need them to be

File: feature-tests/tests/resources/features/aisGET/InvokeApiGateWay-HappyPath.feature [hunk 3]
```diff
@@ -16,6 +16,38 @@ Feature: Invoke-APIGateway-HappyPath.feature
...
+            | suspendNoAction       | unblock                   | false        | AIS_ACCOUNT_SUSPENDED                              | false        | true           | false         | false           |
+            | suspendNoAction       | userActionIdResetSuccess  | false        | AIS_ACCOUNT_SUSPENDED                              | false        | true           | false         | false           |
+            | suspendNoAction       | userActionPswResetSuccess | false        | AIS_ACCOUNT_SUSPENDED                              | false        | true           | false         | false           |
+            | suspendNoAction       | unSuspendAction           | false        | AIS_ACCOUNT_SUSPENDED                              | false        | true           | false         | false           |
+
+    Scenario Outline: Happy Path - Blocked Account - Get Request to /ais/userId - Returns Expected Data for <aisEventType>
+        Given I send an <aisEventType> intervention message to the TxMA ingress SQS queue for a Account in <originalAaisEventType> state
```
Comment (KG700) [inline feature-tests/tests/resources/features/aisGET/InvokeApiGateWay-HappyPath.feature:36]:
```suggestion
        Given I send an <aisEventType> intervention message to the TxMA ingress SQS queue for a Account in <originalAisEventType> state
```
Seems like a typo?

**Review by KG700 — APPROVED** [review]

File: feature-tests/tests/resources/step-definitions/invoke-api-gateway-happy-path.step.ts (hunk 1 above)
Comment (ChrisBates1) [reply feature-tests/tests/resources/step-definitions/invoke-api-gateway-happy-path.step.ts:64]: Good idea Chiara, ill make that change and update the PR, thank you

File: feature-tests/tests/resources/features/aisGET/InvokeApiGateWay-HappyPath.feature (hunk 3 above)
Comment (ChrisBates1) [reply feature-tests/tests/resources/features/aisGET/InvokeApiGateWay-HappyPath.feature:36]: Thanks Karla, ill get this updated in the next PR update

File: feature-tests/tests/resources/step-definitions/invoke-api-gateway-happy-path.step.ts (hunk 1 above)
Comment (ChrisBates1) [reply feature-tests/tests/resources/step-definitions/invoke-api-gateway-happy-path.step.ts:64]: image
changes made and all tests still passing, thank you

File: feature-tests/tests/resources/step-definitions/invoke-api-gateway-happy-path.step.ts (hunk 2 above)
Comment (ChrisBates1) [reply feature-tests/tests/resources/step-definitions/invoke-api-gateway-happy-path.step.ts:106]: image
changes made and all tests still passing, thank you

File: feature-tests/tests/resources/features/aisGET/InvokeApiGateWay-HappyPath.feature (hunk 3 above)
Comment (ChrisBates1) [reply feature-tests/tests/resources/features/aisGET/InvokeApiGateWay-HappyPath.feature]: image
changes made and all tests still passing, thank you

**Pull request merged**